File: app/src/main/java/com/odnovolov/randomuser/utils/FragmentAwareOfFirstViewCreation.java
package com.odnovolov.randomuser.utils;

import android.os.Bundle;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

public class FragmentAwareOfFirstViewCreation extends Fragment {
    private boolean hasSavedInstanceState = false;
    private int numberOfOnViewCreatedInvocation = 0;

    public boolean isViewFirstCreated() {
        return !hasSavedInstanceState && numberOfOnViewCreatedInvocation <= 1;
    }

    @Override
    public void onViewCreated(
            @NonNull View view,
            @Nullable Bundle savedInstanceState
    ) {
        hasSavedInstanceState = savedInstanceState != null;
        numberOfOnViewCreatedInvocation++;
        super.onViewCreated(view, savedInstanceState);
    }
}

File: app/src/main/java/com/odnovolov/randomuser/utils/FragmentAwareOfFirstViewCreation.kt
package com.odnovolov.randomuser.utils

import android.os.Bundle
import android.view.View
import androidx.fragment.app.Fragment

open class FragmentAwareOfFirstViewCreation : Fragment() {
    private var hasSavedInstanceState = false
    private var numberOfOnViewCreatedInvocation = 0

    protected val isViewFirstCreated: Boolean
        get() = !hasSavedInstanceState && numberOfOnViewCreatedInvocation <= 1

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        hasSavedInstanceState = savedInstanceState != null
        numberOfOnViewCreatedInvocation++
        super.onViewCreated(view, savedInstanceState)
    }
}